test/e2e/settings/custom_template.go

```diff
@@ -0,0 +1,752 @@
+					ssl_ciphers                             {{ $server.SSLCiphers }};
+					{{ end }}
+			
+					{{ if not (empty $server.ServerSnippet) }}
+					{{ $server.ServerSnippet }}
+					{{ end }}
+			
+					{{ range $location := $server.Locations }}
+					{{ $path := buildLocation $location }}
+					{{ $proxySetHeader := proxySetHeader $location }}
+					{{ $authPath := buildAuthLocation $location }}
+			
+					{{ if not (empty $location.Rewrite.AppRoot)}}
+					if ($uri = /) {
+						return 302 {{ $location.Rewrite.AppRoot }};
+					}
+					{{ end }}
+			
+					location {{ $path }} {
+					}
+					{{ end }}
+					{{ end }}
+			
+					{{ if eq $server.Hostname "_" }}
+					# health checks in cloud providers require the use of port {{ $all.ListenPorts.HTTP }}
+					location {{ $all.HealthzURI }} {
+						{{ if $all.Cfg.EnableOpentracing }}
+						opentracing off;
+						{{ end }}
+			
+						access_log off;
+						return 200;
+					}
+			
+					# this is required to avoid error if nginx is being monitored
+					# with an external software (like sysdig)
+					location /nginx_status {
+						{{ if $all.Cfg.EnableOpentracing }}
+						opentracing off;
+						{{ end }}
+			
+						{{ range $v := $all.NginxStatusIpv4Whitelist }}
+						allow {{ $v }};
+						{{ end }}
+						{{ if $all.IsIPV6Enabled -}}
+						{{ range $v := $all.NginxStatusIpv6Whitelist }}
+						allow {{ $v }};
+						{{ end }}
+						{{ end -}}
+						deny all;
+			
+						access_log off;
+						stub_status on;
+					}
+			
+					{{ end }}
+			
+			{{ end }}				
+				`,
+			},
+		}
+
+		_, err := f.KubeClientSet.CoreV1().ConfigMaps(f.IngressController.Namespace).Create(configmap)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = framework.UpdateDeployment(f.KubeClientSet, f.IngressController.Namespace, "nginx-ingress-controller", 1,
+			func(deployment *appsv1beta1.Deployment) error {
+				deployment.Spec.Template.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{
+					{
+						Name:      "custom-template",
+						ReadOnly:  true,
+						MountPath: "/etc/nginx/template",
+					},
+				}
+
+				deployment.Spec.Template.Spec.Volumes = []corev1.Volume{
+					{
+						Name: "custom-template",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "custom-template",
+								},
+								Items: []corev1.KeyToPath{
+									{
+										Key:  "nginx.tmpl",
+										Path: "nginx.tmpl",
+									},
+								},
+							},
+						},
+					},
+				}
+
+				_, err := f.KubeClientSet.AppsV1beta1().Deployments(f.IngressController.Namespace).Update(deployment)
+				return err
+			})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should return a custom tempĺate", func() {
+		resp, _, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL).
+			End()
+		Expect(len(errs)).Should(BeNumerically("==", 0))
+		Expect(resp.StatusCode).Should(Equal(404))
```

Comment:
In the template I'd return a custom header or body and assert on it here. Otherwise you aren't asserting that the new template is being used.